c2corg_ui/static/js/documentservice.js

```diff
@@ -124,17 +124,23 @@ app.Document.prototype.hasAssociation = function(type, id) {
  * @param {appx.SimpleSearchDocument} doc
  * @param {string=} doctype Optional doctype
  * @param {boolean=} setOutingTitle
+ * @param {?boolean} editing
  * @export
  */
 app.Document.prototype.pushToAssociations = function(doc, doctype,
-    setOutingTitle) {
+    setOutingTitle, editing) {
   var associations = this.document.associations;
   doctype = doctype || app.utils.getDoctype(doc['type']);
   setOutingTitle = typeof setOutingTitle !== 'undefined' ?
     setOutingTitle : false;
   doc['new'] = true;
   associations[doctype].push(doc);
 
+  // when creating a route, make the first associated wp a main one
+  if (editing && associations.waypoints.length === 1) {
```

Comment:
No I think I misunderstood what you did. I thought you distinguished between editing / creating with this variable.
Because to me, you shouldn't apply the main waypoint automatically to a route you are editing: if the route has no main waypoint, it's a determinate choice. Plus, it is error prone: imagine you only want to correct a sentence in a route without main point. When commiting your changes, you need to uncheck the automatically added waypoint. This is not the expected behaviour, is it?